stream/s2a_stream.go

```diff
@@ -32,3 +34,6 @@ type S2AStream interface {
 	// Closes the channel to the S2A server.
 	CloseSend() error
 }
+
+// GetS2AStream type is for generating an S2AStream interface for talking to the S2A server.
+type GetS2AStream func(ctx context.Context, s2av2Address string, opts ...string) (S2AStream, error)
```

Comment:
do we want to define some `Option` struct here or string tyoe is sufficient?
`opts ...string)`

Reply:
Yeah, I think this allows the greatest flexibility.

If we add an `Option` struct instead, every time if we want to add a new field to this Option we need to create a PR. Instead now we can pass as many parameters as we want and rely on the actual implementation to interpret these parameters.